controllers/candidatesController.js

```diff
@@ -99,28 +98,35 @@ module.exports = {
       work_culture_preferences: req.body.work_culture_preferences,
     };
 
-    Candidate.findByIdAndUpdate(candidateId, { $set: candidateParams })
-      .then(candidate => {
-        res.locals.redirect = `/user/${req.app.locals.user._id}`;
-        res.locals.candidate = candidate;
-        next();
-      })
-      .catch(error => {
-        console.log(`Error updating candidate by ID: ${error.message}`); 
-        next(error);
-      });
+    // Candidate.findByIdAndUpdate(candidateId, { $set: candidateParams })
+    Candidate.findOneAndUpdate({_id: candidateId}, {$set: candidateParams}, {new: true}, (err, job) => {
+        if(err) {
+            req.flash('error', `There has been an error while updating the candidate data: ${error.message}`);
+            console.log(`Error updating candidate by ID: ${error.message}`);
+            next(error);
```

Comment:
interesting why mongoosastic does not accept findbyId, but findOne() only...

Reply:
yea it's weird :/  if you're interested you can check in the part "autoindexing" which functions are supported and which not : https://github.com/mongoosastic/mongoosastic/blob/master/docs/README.md#auto-indexing